In [1]:
#import packages
import pandas as pd

In [2]:
movies_data = pd.read_csv(
    '/Users/malenamalka/Documents/GitHub/preJETwork/data/netflix_data.csv',
    engine='python',
    encoding='utf-8',
    on_bad_lines='skip'
)

movies_data.head()

,Release_Date,Title,Overview,Popularity,Vote_Count,Vote_Average,Original_Language,Genre,Poster_Url
0,2021-12-15,Spider-Man: No Way Home,Peter Parker is unmasked and no longer able to...,5083.954,8940,8.3,en,"Action, Adventure, Science Fiction",https://image.tmdb.org/t/p/original/1g0dhYtq4i...
1,2022-03-01,The Batman,"In his second year of fighting crime, Batman u...",3827.658,1151,8.1,en,"Crime, Mystery, Thriller",https://image.tmdb.org/t/p/original/74xTEgt7R3...
2,2022-02-25,No Exit,Stranded at a rest stop in the mountains durin...,2618.087,122,6.3,en,Thriller,https://image.tmdb.org/t/p/original/vDHsLnOWKl...
3,2021-11-24,Encanto,"The tale of an extraordinary family, the Madri...",2402.201,5076,7.7,en,"Animation, Comedy, Family, Fantasy",https://image.tmdb.org/t/p/original/4j0PNHkMr5...
4,2021-12-22,The King's Man,As a collection of history's worst tyrants and...,1895.511,1793,7.0,en,"Action, Adventure, Thriller, War",https://image.tmdb.org/t/p/original/aq4Pwv5Xeu...


¿Cuántas películas hay en el dataset?
¿Cuántas variables tiene?
¿Qué representa cada variable?
¿Qué tipo de dato debería tener cada columna?
¿Hay columnas que parecen innecesarias?

In [3]:
movies_data.shape

(9827, 9)

El dataset tiene 9827 peliculas, y 9 variables. Las variables son:
- Release_Date: La fecha de estreno de cada pelicula (Fecha)
- Title: Titulo de la pelicula (String)
- Overview: Descripcion de la pelicula (String)
- Popularity: Que tan popular es la pelicula (Number)
- Vote_Count: Cuantas personas votaron que es popular (Int)
- Vote_average: Voto promedio de popularidad (Float)
- Original_Language: Idioma Original (String)
- Genre: Listado de Generos de la pelicula (List of strings)
- Poster_Url: Imagen del poster (string)

In [4]:
#como el poster_url no es importante, lo vamos a droppear

movies_data = movies_data.drop(columns='Poster_Url')
movies_data.head()

,Release_Date,Title,Overview,Popularity,Vote_Count,Vote_Average,Original_Language,Genre
0,2021-12-15,Spider-Man: No Way Home,Peter Parker is unmasked and no longer able to...,5083.954,8940,8.3,en,"Action, Adventure, Science Fiction"
1,2022-03-01,The Batman,"In his second year of fighting crime, Batman u...",3827.658,1151,8.1,en,"Crime, Mystery, Thriller"
2,2022-02-25,No Exit,Stranded at a rest stop in the mountains durin...,2618.087,122,6.3,en,Thriller
3,2021-11-24,Encanto,"The tale of an extraordinary family, the Madri...",2402.201,5076,7.7,en,"Animation, Comedy, Family, Fantasy"
4,2021-12-22,The King's Man,As a collection of history's worst tyrants and...,1895.511,1793,7.0,en,"Action, Adventure, Thriller, War"


¿Hay valores faltantes?
¿Hay registros duplicados?

In [5]:
#check for nulls:
print(movies_data.isna().sum())
#no hay valores faltantes

Release_Date         0
Title                0
Overview             0
Popularity           0
Vote_Count           0
Vote_Average         0
Original_Language    0
Genre                0
dtype: int64


In [6]:
#check for duplicates:
duplicates = movies_data[movies_data.duplicated()]
print(duplicates)
#no hay duplicates

Empty DataFrame
Columns: [Release_Date, Title, Overview, Popularity, Vote_Count, Vote_Average, Original_Language, Genre]
Index: []


In [7]:
#desarmar listas de generos:
max_genres = movies_data['Genre'].str.split(', ').str.len().max()

genre_columns = [f'Genre_{i}' for i in range(1, max_genres + 1)]

movies_data[genre_columns] = pd.DataFrame(
    movies_data['Genre'].str.split(', ').tolist(),
    index=movies_data.index
)

movies_data.head(10)

,Release_Date,Title,Overview,Popularity,Vote_Count,Vote_Average,Original_Language,Genre,Genre_1,Genre_2,Genre_3,Genre_4,Genre_5,Genre_6,Genre_7,Genre_8
0,2021-12-15,Spider-Man: No Way Home,Peter Parker is unmasked and no longer able to...,5083.954,8940,8.3,en,"Action, Adventure, Science Fiction",Action,Adventure,Science Fiction,None,None,None,None,None
1,2022-03-01,The Batman,"In his second year of fighting crime, Batman u...",3827.658,1151,8.1,en,"Crime, Mystery, Thriller",Crime,Mystery,Thriller,None,None,None,None,None
2,2022-02-25,No Exit,Stranded at a rest stop in the mountains durin...,2618.087,122,6.3,en,Thriller,Thriller,None,None,None,None,None,None,None
3,2021-11-24,Encanto,"The tale of an extraordinary family, the Madri...",2402.201,5076,7.7,en,"Animation, Comedy, Family, Fantasy",Animation,Comedy,Family,Fantasy,None,None,None,None
4,2021-12-22,The King's Man,As a collection of history's worst tyrants and...,1895.511,1793,7.0,en,"Action, Adventure, Thriller, War",Action,Adventure,Thriller,War,None,None,None,None
5,2022-01-07,The Commando,An elite DEA agent returns home after a failed...,1750.484,33,6.6,en,"Action, Crime, Thriller",Action,Crime,Thriller,None,None,None,None,None
6,2022-01-12,Scream,Twenty-five years after a streak of brutal mur...,1675.161,821,6.8,en,"Horror, Mystery, Thriller",Horror,Mystery,Thriller,None,None,None,None,None
7,2022-02-10,Kimi,A tech worker with agoraphobia discovers recor...,1601.782,206,6.3,en,Thriller,Thriller,None,None,None,None,None,None,None
8,2022-02-17,Fistful of Vengeance,A revenge mission becomes a fight to save the ...,1594.013,114,5.3,en,"Action, Crime, Fantasy",Action,Crime,Fantasy,None,None,None,None,None
9,2021-11-03,Eternals,The Eternals are a team of ancient aliens who ...,1537.406,4726,7.2,en,Science Fiction,Science Fiction,None,None,None,None,None,None,None


Parte 3 — Análisis univariado

¿Cuáles son los idiomas más frecuentes?

In [8]:
#idiomas:
movies_data['Original_Language'].value_counts().head(5)

en    7570
ja     645
es     339
fr     292
ko     170
Name: Original_Language, dtype: int64

Los idiomas mas frecuentes son inles, japones, español, frances y koreano
¿Qué porcentaje del dataset está en inglés?

In [9]:
movies_data['Original_Language'].value_counts(normalize = True).head(1)

en    0.770327
Name: Original_Language, dtype: float64

77% de las peliculas estan en ingles
¿Cuáles son los 10 géneros más frecuentes?

In [10]:
movies_data['Genre_1'].value_counts().head(10)

Drama        1791
Action       1570
Comedy       1484
Horror        868
Animation     805
Adventure     586
Thriller      515
Crime         391
Family        350
Romance       304
Name: Genre_1, dtype: int64

¿Cuáles son las películas con mayor popularidad?

In [11]:
movies_data.sort_values(by='Popularity', ascending = False).head(5)

,Release_Date,Title,Overview,Popularity,Vote_Count,Vote_Average,Original_Language,Genre,Genre_1,Genre_2,Genre_3,Genre_4,Genre_5,Genre_6,Genre_7,Genre_8
0,2021-12-15,Spider-Man: No Way Home,Peter Parker is unmasked and no longer able to...,5083.954,8940,8.3,en,"Action, Adventure, Science Fiction",Action,Adventure,Science Fiction,None,None,None,None,None
1,2022-03-01,The Batman,"In his second year of fighting crime, Batman u...",3827.658,1151,8.1,en,"Crime, Mystery, Thriller",Crime,Mystery,Thriller,None,None,None,None,None
2,2022-02-25,No Exit,Stranded at a rest stop in the mountains durin...,2618.087,122,6.3,en,Thriller,Thriller,None,None,None,None,None,None,None
3,2021-11-24,Encanto,"The tale of an extraordinary family, the Madri...",2402.201,5076,7.7,en,"Animation, Comedy, Family, Fantasy",Animation,Comedy,Family,Fantasy,None,None,None,None
4,2021-12-22,The King's Man,As a collection of history's worst tyrants and...,1895.511,1793,7.0,en,"Action, Adventure, Thriller, War",Action,Adventure,Thriller,War,None,None,None,None


¿Cuáles tienen mejor puntuación promedio?

In [12]:
movies_data.sort_values(by='Vote_Average', ascending = False).head(5)

,Release_Date,Title,Overview,Popularity,Vote_Count,Vote_Average,Original_Language,Genre,Genre_1,Genre_2,Genre_3,Genre_4,Genre_5,Genre_6,Genre_7,Genre_8
9391,2020-04-09,Kung Fu Master Huo Yuanjia,The young and vigorous Huo Yuanjia was only fo...,13.745,1,10.0,zh,"Action, Drama",Action,Drama,None,None,None,None,None,None
7339,2018-06-08,Franco Escamilla: Por La Anécdota,Mexican stand-up comedian Franco Escamilla dra...,16.158,92,9.2,es,Comedy,Comedy,None,None,None,None,None,None,None
667,2019-03-29,Demon Slayer: Kimetsu no Yaiba Sibling's Bond,Tanjiro finds his family slaughtered and the l...,86.689,27,9.1,ja,"Action, Animation, Fantasy",Action,Animation,Fantasy,None,None,None,None,None
2325,2021-06-17,Impossible Things,A widow who is tormented by the memory of her ...,36.466,82,9.1,es,"Family, Drama",Family,Drama,None,None,None,None,None,None
7401,2020-10-30,My Sex Doll,The story of chaos both body and mind occurs w...,16.088,4,9.0,th,"Comedy, Romance",Comedy,Romance,None,None,None,None,None,None


¿Cuáles tienen mayor cantidad de votos?

In [13]:
movies_data.sort_values(by='Vote_Count', ascending = False).head(5)

,Release_Date,Title,Overview,Popularity,Vote_Count,Vote_Average,Original_Language,Genre,Genre_1,Genre_2,Genre_3,Genre_4,Genre_5,Genre_6,Genre_7,Genre_8
380,2010-07-15,Inception,"Cobb, a skilled thief who commits corporate es...",127.131,31077,8.4,en,"Action, Science Fiction, Adventure",Action,Science Fiction,Adventure,None,None,None,None,None
461,2014-11-05,Interstellar,The adventures of a group of explorers who mak...,110.531,27975,8.4,en,"Adventure, Drama, Science Fiction",Adventure,Drama,Science Fiction,None,None,None,None,None
221,2008-07-14,The Dark Knight,Batman raises the stakes in his war on crime. ...,173.923,26994,8.5,en,"Drama, Action, Crime, Thriller",Drama,Action,Crime,Thriller,None,None,None,None
138,2012-04-25,The Avengers,When an unexpected enemy emerges and threatens...,229.772,26490,7.7,en,"Science Fiction, Action, Adventure",Science Fiction,Action,Adventure,None,None,None,None,None
195,2016-02-09,Deadpool,Deadpool tells the origin story of former Spec...,189.206,26390,7.6,en,"Action, Adventure, Comedy",Action,Adventure,Comedy,None,None,None,None,None


¿Cuál es la popularidad promedio por idioma? 

In [18]:
pop_by_language = movies_data.groupby('Original_Language')['Popularity'].mean()
pop_by_language

Original_Language
ar     17.296000
bn     18.059000
ca     17.214000
cn     23.140780
cs     16.701500
da     22.421821
de     25.943000
el     16.509600
en     41.759701
es     37.969496
et     13.822000
eu     21.859000
fa     19.227667
fi     39.041400
fr     32.921370
he     14.743500
hi     85.931654
hu     14.394667
id     26.699533
is     24.312500
it     25.889862
ja     40.723309
ko     33.513265
la     17.138000
lv     21.540000
ml    151.512000
ms     18.416000
nb     22.410000
nl     24.479238
no     27.515462
pl     53.809882
pt     34.459189
ro     30.179500
ru     39.658193
sr     19.190800
sv     21.161087
ta     15.726500
te     30.460333
th     60.870059
tl     21.156875
tr     71.631400
uk     19.581500
zh     30.019581
Name: Popularity, dtype: float64

¿Cuál es el rating promedio por idioma?

¿Qué idioma tiene películas con más votos en promedio?

 ¿Qué géneros tienen mejor puntuación promedio?

¿Qué géneros tienen mayor popularidad promedio? 

¿Qué géneros generan más engagement (Vote_Count)?